search/django/weedid/models.py

```diff
@@ -1,3 +1,43 @@
-# from django.db import models
+from django.db import models
+from django.contrib.postgres.fields import JSONField
+from django.contrib.auth.models import AbstractUser
 
-# Create your models here.
+
+class Dataset(models.Model):
+    UPLOAD_STATUS_CHOICES = [
+        ("N", "None"),
+        ("P", "Processing"),
+        ("I", "Incomplete"),
+        ("AR", "Awaiting Review"),
+        ("F", "Failed"),
+        ("C", "Complete"),
+    ]
+    metadata = JSONField(null=True)
+    upload_id = models.CharField(
```

Comment:
If we essentially want an atomic auto-increment situation, why don't we use the automatic ID? Or is this to make sure that the upload IDs are unique with respect to the stored repository, not just what's in the database?